### PR TITLE
feat(service-portal): Add properties section to minar-sidur

### PR DESCRIPTION
**Pull request by thordurhhh**
## What

Add properties service.
Add properties section to minarsidur.
This will be released behind a feature flag.

## Screenshots / Gifs

Properties

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/service-portal/assets/src/screens/AssetsOverview/AssetsOverview.tsx [hunk 1]
```diff
@@ -1,32 +1,92 @@
...
+          title={defineMessage({
+            id: 'sp.assets:title',
+            defaultMessage: 'Fasteignir',
+          })}
+          intro={defineMessage({
+            id: 'sp.assets:intro',
+            defaultMessage:
+              'Hér færðu upplýsingar úr fasteignaskrá um fasteignir þínar, lönd og lóðir sem þú ert skráður eigandi að.',
+          })}
+          img="./assets/images/educationGrades.svg"
```
Comment (disaerna) [inline libs/service-portal/assets/src/screens/AssetsOverview/AssetsOverview.tsx:73]: Er þetta rétt mynd? 🤔

File: libs/service-portal/assets/src/screens/RealEstateAssetDetail/RealEstateAssetDetail.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+          title={defineMessage({
+            id: 'sp.assets:title',
+            defaultMessage: 'Fasteignir',
+          })}
+          intro={defineMessage({
+            id: 'sp.assets:intro',
+            defaultMessage:
+              'Hér færðu upplýsingar úr fasteignaskrá um fasteignir þínar, lönd og lóðir sem þú ert skráður eigandi að.',
+          })}
+          img="./assets/images/educationGrades.svg"
```
Comment (disaerna) [inline libs/service-portal/assets/src/screens/RealEstateAssetDetail/RealEstateAssetDetail.tsx:116]: Þessi mynd hér aftur

File: libs/api/domains/assets/src/lib/api-domains-assets.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import { AssetsXRoadService } from './api-domains-assets.service'
+
+@UseGuards(IdsAuthGuard, IdsUserGuard, ScopesGuard)
+@Audit({ namespace: '@island.is/api/assets' })
+export class AssetsXRoadResolver {
+  constructor(private assetsXRoadService: AssetsXRoadService) {}
+
+  @Query(() => graphqlTypeJson)
+  @Audit()
+  async getRealEstates(
```
Comment (darrikonn) [inline libs/api/domains/assets/src/lib/api-domains-assets.resolver.ts:28]: The GraphQL names are not following the island.is naming conventions:
https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions

Comment (darrikonn) [reply libs/api/domains/assets/src/lib/api-domains-assets.resolver.ts:28]: queries/types/inputs etc.

File: libs/service-portal/assets/src/screens/RealEstateAssetDetail/RealEstateAssetDetail.tsx [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+                      )
+                    : '',
+                ],
+              ],
+            },
+          ]}
+        />
+      </Box>
+      <Box marginTop={7}>
+        {assetData?.notkunareiningar?.notkunareiningar &&
```
Comment (disaerna) [inline libs/service-portal/assets/src/screens/RealEstateAssetDetail/RealEstateAssetDetail.tsx:164]: Er nauðsynlegt að gera && þegar það er verið að nota spurningarmerki?

File: libs/api/domains/assets/src/models/propertyOwners.model.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  hasPreviousPage?: boolean
+
+  @Field({ nullable: true })
+  hasNextPage?: boolean
+}
+
+@ObjectType()
+export class PropertyOwner {
+  @Field({ nullable: true })
+  nafn?: string | undefined
```
Comment (darrikonn) [inline libs/api/domains/assets/src/models/propertyOwners.model.ts:30]: Should the fields not be in English?

Comment (darrikonn) [reply libs/api/domains/assets/src/models/propertyOwners.model.ts:30]: ref: https://docs.devland.is/technical-overview/code-standards#language

**Review by disaerna — COMMENTED** [review]
Nokkrar litlar athugasemdir í ServicePortal, annars bara LGTM 😃

File: libs/service-portal/assets/src/components/AssetDisclaimer/index.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import { messages } from '../../lib/messages'
+
+const AssetDisclaimer = () => {
+  const { formatMessage } = useLocale()
+  return (
+    <Box>
+      <p style={{ marginTop: 32, fontWeight: 100 }}>
+        {formatMessage(messages.disclaimerA)}
+      </p>
+      <p style={{ marginTop: 24, fontWeight: 100 }}>
```
Comment (darrikonn) [inline libs/service-portal/assets/src/components/AssetDisclaimer/index.tsx:13]: Reason for not using `<Text />`?

File: libs/service-portal/assets/src/components/TableUnits/index.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+                    alignItems="center"
+                    justifyContent="center"
+                    display="flex"
+                  >
+                    <Button
+                      size="small"
+                      variant="text"
+                      onClick={() => getMoreItems()}
+                    >
+                      Sækja meira
```
Comment (darrikonn) [inline libs/service-portal/assets/src/components/TableUnits/index.tsx:63]: translations

**Review by darrikonn — CHANGES_REQUESTED** [review]
Main reason for Request Changes is GraphQL Naming Conventions + English vs. Icelandic

**Review by valurefugl — APPROVED** [review]

File: libs/service-portal/assets/src/components/AssetDisclaimer/index.tsx (hunk 6 above)
Comment (thordurhhh) [reply libs/service-portal/assets/src/components/AssetDisclaimer/index.tsx:13]: Whoops. I'll fix that.

File: libs/auth-nest-tools/src/lib/auth.middleware.ts [hunk 8]
```diff
@@ -56,6 +57,9 @@ export class AuthMiddleware implements Middleware {
 
     context.init.headers = Object.assign({}, context.init.headers, {
       authorization: bearerToken,
+      ...(this.options.authID && {
+        ['Authorization-Identity']: bearerToken,
+      }),
```
Comment (eirikurn) [inline libs/auth-nest-tools/src/lib/auth.middleware.ts:62]: This looks very non-standard. Where does this come from?

Would be good to document. If this is needed in a single integration, we should just do it in another middleware, or add a more flexible option to this middleware: `headerName = 'authorization'`.

Comment (thordurhhh) [reply libs/auth-nest-tools/src/lib/auth.middleware.ts:62]: I believe this is a temporary solution from the providers of the Fasteignir API and they will update it eventually to require something else than the same bearer token as the Authorization.
But for now they requested it to be the same.
I'm not sure if this is just a one off for `Fasteignir-v1` or if they will require this for all `SKRA-Protected` in the coming future.

**Review by disaerna — APPROVED** [review]
Nice 👍

File: libs/auth-nest-tools/src/lib/auth.middleware.ts (hunk 8 above)
Comment (eirikurn) [reply libs/auth-nest-tools/src/lib/auth.middleware.ts:66]: Ok.

Then can we move this code somewhere else? :)

Comment (eirikurn) [reply libs/auth-nest-tools/src/lib/auth.middleware.ts:66]: cc @vigfusg @saevarma. I always get worried when I see non-standard security logic. 😅

Should we ask for more information about what they need this non-standard header for? Why the access token is not enough?

**Pull request merged**